tests/e2e/hcp_machine_pool_test.go

```diff
@@ -1,23 +1,100 @@
 package e2e
 
 import (
+	"fmt"
 	"strings"
 
 	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
 	"github.com/openshift-online/ocm-common/pkg/test/vpc_client"
 
 	"github.com/openshift/rosa/tests/ci/labels"
+	"github.com/openshift/rosa/tests/utils/common"
+	"github.com/openshift/rosa/tests/utils/common/constants"
+	"github.com/openshift/rosa/tests/utils/config"
 	"github.com/openshift/rosa/tests/utils/exec/rosacli"
 	ph "github.com/openshift/rosa/tests/utils/profilehandler"
 )
 
-var _ = Describe("Create Machine Pool", labels.Feature.Machinepool, func() {
+var _ = Describe("HCP Machine Pool", labels.Feature.Machinepool, func() {
+	// It doesn't check whether node pool instances ready in default.
+	// If needed for verify hcp node pool's changes, pls set the ENV CLUSTER_NODE_POOL_GLOBAL_CHECK to true,
+	// which will wait for node pool instances ready until timeout.
+	isNodePoolGlobalCheck := config.IsNodePoolGlobalCheck()
 
 	var (
-		rosaClient *rosacli.Client
-		profile    *ph.Profile
+		rosaClient  *rosacli.Client
+		profile     *ph.Profile
+		isMultiArch bool
 	)
+
+	checkReplicas := func(mpName string, isAutoscale bool) bool {
+		if isAutoscale {
+			replicas, err := rosaClient.MachinePool.GetNodePoolAutoScaledReplicas(clusterID, mpName)
+			if err != nil {
+				return false
+			}
+			return replicas["Current replicas"] == replicas["Min replicas"]
+
+		} else {
+			mpDesc, err := rosaClient.MachinePool.DescribeAndReflectNodePool(clusterID, mpName)
+			if err != nil {
+				return false
+			}
+			return mpDesc.CurrentReplicas == fmt.Sprintf("%v", mpDesc.DesiredReplicas)
+		}
+	}
+
+	scaleNodePool := func(mpName string, scaleFlag string, updateReplicas int, waitForNPInstancesReady bool) {
+		_, err := rosaClient.MachinePool.EditMachinePool(clusterID, mpName,
+			"--replicas", fmt.Sprintf("%v", updateReplicas),
+			"-y",
+		)
+		Expect(err).ToNot(HaveOccurred())
+
+		By("Check the machinepool replicas after scale " + scaleFlag)
+		mpDesc, err := rosaClient.MachinePool.DescribeAndReflectNodePool(clusterID, mpName)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(mpDesc.DesiredReplicas).Should(Equal(updateReplicas))
+
+		if waitForNPInstancesReady && isNodePoolGlobalCheck {
```

Comment:
We don't wait for instances ready in rosacli cases. Please put this part to backend. It's not related to rosacli if the instance type can work to the nodepool or not. Just make sure the rosacli is acting well with backend

Reply:
Hmm, I indeed added it for `rosacli cases`. For an extreme example, if the nodes are ready from CMS, but it may  show wrong `Current Replicas` or `Desired Replicas` due to `rosacli issue`, so I added this logic, but it won't check in default. It only checks when setting the ENV `export CLUSTER_NODE_POOL_GLOBAL_CHECK=true` if needed.

Reply:
If that is the checkpoint let's just make sure `Desired Replicas= <nodes in backend>` and at the same time `Current Replicas=<desired nodes in the response body>`